Python/compile.c

```diff
@@ -5421,6 +5452,601 @@ compiler_slice(struct compiler *c, expr_ty s)
     return 1;
 }
 
+
+// PEP 634: Structural Pattern Matching
+
+// To keep things simple, all compiler_pattern_* routines follow the convention
+// of preserving TOS (the subject for the given pattern) and pushing either True
+// (match) or False (no match) on top of it. We do this even for irrefutable
+// patterns; the idea is that it's much easier to smooth out any redundant
+// pushing, popping, and jumping in the peephole optimizer than to detect or
+// predict it here.
+
+
+#define WILDCARD_CHECK(N) \
+    ((N)->kind == Name_kind && \
+    _PyUnicode_EqualToASCIIString((N)->v.Name.id, "_"))
+
+
+static int
+pattern_helper_load_attr(struct compiler *c, expr_ty p, pattern_context *pc)
+{
+    assert(p->kind == Attribute_kind);
+    assert(p->v.Attribute.ctx == Load);
+    if (p->v.Attribute.value->kind == Attribute_kind) {
+        RETURN_IF_FALSE(pattern_helper_load_attr(c, p->v.Attribute.value, pc));
+    }
+    else {
+        assert(p->v.Attribute.value->kind == Name_kind);
+        assert(p->v.Attribute.value->v.Name.ctx == Load);
+        RETURN_IF_FALSE(compiler_nameop(c, p->v.Attribute.value->v.Name.id, Load));
+    }
+    ADDOP_NAME(c, LOAD_ATTR, p->v.Attribute.attr, names);
+    return 1;
+}
+
+
+static int
+pattern_helper_load_constant(struct compiler *c, expr_ty p, pattern_context *pc)
```

Comment:
Replace uses of this function with `ADDOP_LOAD_CONST(c, p->v.Constant.value);`
The asserts should have been checked in the verification step. Aren't they?